cuda/solver/lower_trs_kernels.cu

```diff
@@ -69,20 +69,16 @@ void should_perform_transpose(std::shared_ptr<const CudaExecutor> exec,
 }
 
 
-void init_struct(std::shared_ptr<const CudaExecutor> exec,
-                 std::shared_ptr<solver::SolveStruct>& solve_struct)
-{
-    init_struct_kernel(exec, solve_struct);
-}
-
-
 template <typename ValueType, typename IndexType>
 void generate(std::shared_ptr<const CudaExecutor> exec,
               const matrix::Csr<ValueType, IndexType>* matrix,
-              solver::SolveStruct* solve_struct, const gko::size_type num_rhs)
+              std::shared_ptr<solver::SolveStruct>& solve_struct,
+              const gko::size_type num_rhs)
 {
-    generate_kernel<ValueType, IndexType>(exec, matrix, solve_struct, num_rhs,
-                                          false);
+    if (matrix->get_strategy()->get_name() == "sparselib") {
```

Comment:
There was some talk earlier about overhauling the strategy system. Are we are keeping it as it is? I guess it does sound like a good idea to have the strategy control not only the spmv but also the trsv.

Reply:
We were planning on using a factory parameter instead of the strategy, so yes, that makes sense.